src/Administration/Resources/app/administration/src/core/factory/service.factory.ts

```diff
@@ -5,7 +5,7 @@
 type ServiceObject = {
     get: <SN extends keyof ServiceContainer>(serviceName: SN) => ServiceContainer[SN];
     list: () => (keyof ServiceContainer)[];
-    register: <SN extends keyof ServiceContainer>(serviceName: SN, service: ServiceContainer[SN]) => void;
+    register: typeof Shopware.Application.addServiceProvider;
```

Comment:
I am confused with this. How is it secured that the correct service name and container will be used then?

Reply:
Because the generic is already part of `Shopware.Application.addServiceProvider`